web-api/src/applicationContext.js

```diff
@@ -1275,7 +1266,15 @@ const gatewayMethods = {
 };
 
 module.exports = appContextUser => {
-  if (appContextUser) setCurrentUser(appContextUser);
+  let user;
+
+  if (appContextUser) {
+    user = new User(appContextUser);
+  }
+
+  const getCurrentUser = () => {
+    return user;
+  };
```

Comment:
Ooh, one more thing to mention — the `user` associated with an `applicationContext` was previously stored in a module-scoped variable. This meant that:

```js
const appContext1 = createApplicationContext(user1);
appContext1.getCurrentUser(); // returns user1

const appContext2 = createApplicationContext(user2);
appContext2.getCurrentUser(); // returns user2

// has been modified by the second function, now returns user2
appContext1.getCurrentUser(); // returns user2
```

There aren’t tests for the applicationContext, but I moved it to a function-scoped variable here so it no longer interacts with other instances.